cmd/node/main.go

```diff
@@ -542,6 +552,70 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		log.Debug("no epoch db found in storage", "error", errNotCritical.Error())
 	}
 
+	epochFoundInStorage := errNotCritical == nil
+
+	shouldCallEpochStartDataProvider := bootstrap.ShouldSyncWithTheNetwork(
+		startTime,
+		epochFoundInStorage,
+		nodesConfig,
+		generalConfig,
+	)
+	shouldCallEpochStartDataProvider = true
+	if shouldCallEpochStartDataProvider {
```

Comment:
shouldCall has to be resolved inside the epochStartData provider. Suggestion is to have a factory for the epoch start data provider, launch it in any case, and it resolves dependent on all the cases discussed. Main should be much more simple.

Reply:
done. for the situations when sync is not needed, the factory will return a disabled epoch start data provider which will return an error for Boostrap(), so fresh start